lib/trip.rb

```diff
@@ -1,41 +1,71 @@
 require 'csv'
+require 'time'
 
 require_relative 'csv_record'
 
 module RideShare
   class Trip < CsvRecord
-    attr_reader :id, :passenger, :passenger_id, :start_time, :end_time, :cost, :rating
+    attr_reader :id, :passenger, :passenger_id, :start_time, :end_time, :cost, :rating, :driver_id, :driver
 
     def initialize(
           id:,
+          driver: nil,
+          driver_id: nil,
           passenger: nil,
           passenger_id: nil,
           start_time:,
-          end_time:,
+          end_time: nil,
           cost: nil,
-          rating:
+          rating: nil
         )
       super(id)
 
+      
+      if driver
+        @driver = driver
+        @driver_id = driver.id
+      elsif driver_id
+        @driver_id = driver_id
+      else
+        raise ArgumentError, 'Driver or driver_id is required'
+      end
+
+
       if passenger
         @passenger = passenger
         @passenger_id = passenger.id
-
       elsif passenger_id
         @passenger_id = passenger_id
-
       else
         raise ArgumentError, 'Passenger or passenger_id is required'
       end
 
+
+      if start_time.class == String 
+        start_time = Time.parse(start_time)
+      end
       @start_time = start_time
+
+      if end_time.class == String 
+        end_time = Time.parse(end_time)
+      end
       @end_time = end_time
+      
       @cost = cost
       @rating = rating
 
-      if @rating > 5 || @rating < 1
+      if @rating && (@rating > 5 || @rating < 1)
         raise ArgumentError.new("Invalid rating #{@rating}")
       end
+
+      if @end_time && (end_time < start_time)
+        raise ArgumentError.new("End time #{end_time} is before  start time #{start_time}")
+      end
+    end 
+
+    def trip_duration_in_seconds
+      trip_duration = @end_time - @start_time
+      return trip_duration
```

Comment:
You don't need to assign your results to variables before returning them:

```suggestion
      return @end_time - @start_time
```